src/resolveGitDir.js

```diff
@@ -1,7 +1,10 @@
 'use strict'
 
-const findParentDir = require('find-parent-dir')
+const childProcess = require('child_process')
 
-module.exports = function resolveGitDir() {
-  return findParentDir.sync(process.cwd(), '.git')
-}
+const gitDir = childProcess
```

Comment:
1. We're using `execa` in this project to ensure compatibility between different OSes and also remove chores like converting to UTF etc.
1. There is a `execGit` wrapper already in this project, which utilizes execa. Can it be used here instead so we have the abstraction in one place?

Reply:
I'd prefer not use `sync` and use promises instead TBH. This way it's non-blocking. I also think it's a better design than exporting a constant from the module since it's an implicit behavior and might break if we start moving things around. I'd say, let's keep previous design since I don't think there is a perf problem with it. If we found it's too slow, we could implement a caching layer on top.

Reply:
`runAll`

Reply:
Turning to promises will lead to a lot of changes of original code. `runAll`, `generateTasks` and `makeCmdTasks` have to be promises, which means refactor of many original test cases.

Using `execa` will also fail some test, since some cases write in `expect(execa).toHaveBeenCalledWith`.

I have update my implementation, you can check whether I should work on.

Reply:
So my question is if the fact we depend so much on this function is a code smell. It looks like we should be passing the value in params instead